include/verilated_random.cpp

```diff
@@ -480,12 +451,34 @@ bool VlRandomizer::parseSolution(std::iostream& f) {
         if (m_randmode && !varr.randModeIdxNone()) {
             if (!(m_randmode->at(varr.randModeIdx()))) continue;
         }
-        if (indices.size() > 1) {
-            const std::string flattenedIndex = flattenIndices(indices, &varr);
-            varr.set(flattenedIndex, value);
-        } else {
-            varr.set(idx, value);
+        if (!indices.empty()) {
+            std::ostringstream oss;
+            oss << varr.name();
+            for (const auto& hex_index : indices) {
+                size_t start = hex_index.find_first_not_of(" ");
+                if (start == std::string::npos || hex_index.substr(start, 2) != "#x") {
+                    VL_WARN_MT(__FILE__, __LINE__, "randomize",
+                               "Warning: hex_index contains invalid format");
```

Comment:
Can this ever really happen?  If so, need a test for it.  If not, is VL_FATAL_MT is more appropriate?  Similar below.